pixeltable/share/packager.py

```diff
@@ -225,9 +210,137 @@ def __build_tarball(self) -> Path:
         with tarfile.open(bundle_path, 'w:bz2') as tf:
             # Add metadata json
             tf.add(self.tmp_dir / 'metadata.json', arcname='metadata.json')
-            # Add the Iceberg warehouse dir (including the catalog)
-            tf.add(self.tmp_dir / 'warehouse', arcname='warehouse', recursive=True)
+            # Add the dir containing Parquet tables
+            tf.add(self.tables_dir, arcname='tables')
             # Add the media files
             for src_file, dest_name in self.media_files.items():
                 tf.add(src_file, arcname=f'media/{dest_name}')
         return bundle_path
+
+
+class TableRestorer:
+    tbl_path: str
+    md: Optional[dict[str, Any]]
+    tmp_dir: Path
+    media_files: dict[str, str]  # Mapping from pxtmedia:// URLs to local file:// URLs
+
+    def __init__(self, tbl_path: str, md: Optional[dict[str, Any]] = None) -> None:
+        self.tbl_path = tbl_path
+        self.md = md
+        self.tmp_dir = Path(Env.get().create_tmp_path())
+        self.media_files = {}
+
+    def restore(self, bundle_path: Path) -> pxt.Table:
+        # Extract tarball
+        print(f'Extracting table data into: {self.tmp_dir}')
+        with tarfile.open(bundle_path, 'r:bz2') as tf:
+            tf.extractall(path=self.tmp_dir)
+
+        if self.md is None:
+            # No metadata supplied; read it from the archive
+            with open(self.tmp_dir / 'metadata.json', 'r', encoding='utf8') as fp:
+                self.md = json.load(fp)
+
+        pxt_md_version = self.md['pxt_md_version']
+        assert isinstance(pxt_md_version, int)
+
+        if pxt_md_version != metadata.VERSION:
+            raise excs.Error(
+                f'Pixeltable metadata version mismatch: {pxt_md_version} != {metadata.VERSION}.\n'
+                'Please upgrade Pixeltable to use this dataset: pip install -U pixeltable'
+            )
+
+        tbl_md = [schema.FullTableMd.from_dict(t) for t in self.md['md']['tables']]
+
+        # Create the replica table
+        replica_tbl = catalog.Catalog.get().create_replica(catalog.Path(self.tbl_path), tbl_md)
+        assert replica_tbl._tbl_version.get().is_snapshot
+
+        # Now we need to instantiate and load data for replica_tbl and its ancestors, except that we skip
+        # replica_tbl itself if it's a pure snapshot.
+        if replica_tbl._id != replica_tbl._tbl_version.id:
```

Comment:
Better test that doesn't use internals:
isinstance(View) and replica_tbl.snapshot_only (requires adding a View.snapshot_only property)